cli_tools/gce_windows_upgrade/upgrader/validators.go

```diff
@@ -200,21 +210,23 @@ func validateAndDeriveOSDisk(osDisk *compute.AttachedDisk, derivedVars *derivedV
 	return nil
 }
 
-func validateLicense(osDisk *compute.AttachedDisk, sourceOS string) error {
+func validateLicense(osDisk *compute.AttachedDisk, sourceOS, targetOS string) error {
 	matchSourceOSVersion := false
 	upgraded := false
 	for _, lic := range osDisk.Licenses {
-		if strings.HasSuffix(lic, expectedCurrentLicense[sourceOS]) {
-			matchSourceOSVersion = true
-		} else if strings.HasSuffix(lic, licenseToAdd[sourceOS]) {
-			upgraded = true
+		for _, expectedLic := range upgradePaths[sourceOS][targetOS].expectedCurrentLicense {
+			if strings.HasSuffix(lic, expectedLic) {
+				matchSourceOSVersion = true
+			} else if strings.HasSuffix(lic, upgradePaths[sourceOS][targetOS].licenseToAdd) {
+				upgraded = true
+			}
 		}
 	}
 	if !matchSourceOSVersion {
-		return daisy.Errf(fmt.Sprintf("Can only upgrade GCE instance with %v license attached", expectedCurrentLicense[sourceOS]))
+		return daisy.Errf(fmt.Sprintf("Can only upgrade GCE instance with any of %v license attached", upgradePaths[sourceOS][targetOS].expectedCurrentLicense))
 	}
 	if upgraded {
-		return daisy.Errf(fmt.Sprintf("The GCE instance is with %v license attached, which means it either has been upgraded or has started an upgrade in the past.", licenseToAdd[sourceOS]))
+		return daisy.Errf(fmt.Sprintf("The GCE instance is with %v license attached, which means it either has been upgraded or has started an upgrade in the past.", upgradePaths[sourceOS][targetOS].licenseToAdd))
```

Comment:
The GCE instance has the %v license attached. This likely means the instance has already been upgraded.

Reply:
adopted this words but still keeps "started an upgrade"